### ISSUE TITLE
Empty lines in comments not handled sensibly by require-starting-space

### ISSUE THREAD

**Issue by g1ps** [adrienverge/yamllint#171]
**Problem**
Comments like this:
```
[Line 32 of yml file]
      # - name: some task
      #   import_role:
      #     name: some-role
      #
      # - name: some task
      #   import_role:
      #     name: some-role
      #
      # - name: some task
      #   import_role:
      #     name: some-role
      #
      # - name: some task
      #   import_role:
      #     name: some-role
```
are flagged as needing a leading space after the hash.

```
$ yamllint ./zz.yml
./zz.yml
  36:8      warning  missing starting space in comment  (comments)
  40:8      warning  missing starting space in comment  (comments)
  44:8      warning  missing starting space in comment  (comments)
```

**Request**
That makes no sense as it's likely to flag various valid comments. The rule should only be applied to lines with text following the hash. It should not be applied to cases where a newline follows the hash.

Thanks.

**Comment by adrienverge** [comment]
The problems you describe don't happen on my side.

I suspect interference with potential strange line-endings on Windows. Do you have the `new-lines` rule enabled? Do your files end with universal (`\n`) newlines?

**Comment by g1ps** [comment]
Note that neither:
```
  # yamllint: disable
...
  # yamllint: enable
```
nor
```
  # yamllint: disable rule:comments
...
  # yamllint: enable rule:comments
```
nor
```
# yamllint: disable rule:require-starting-space
...
# yamllint: enable rule:require-starting-space
```
work. In fact I can't disable rules at all for any I've tested so far. `yamllint` even picks up hte indentation of the disabling comments if they're not indented like the content, which seems dubious. Again the docs aren't detailed enough to tell me whether I can disable a rule parameter this way (I should certainly be able to) or not.

**Comment by g1ps** [comment]
> I suspect interference with potential strange line-endings on Windows. Do you have the `new-lines` rule enabled? Do your files end with universal (`\n`) newlines?

No, it's disabled since we might have either. They end with Windows line endings here.

Thanks.

**Comment by g1ps** [comment]
You're right. I ran dos2unix on a file which showed these problems and they went away. This will be a problem for a lot of users trying to run this way to avoid pain with Ansible. Even Linux users might validly have files with Windows line endings on their machines so I hope you can add checking for both types.

Disabling rules works now, too, except that indentation applies to the comments so it expects this:
```
  # yamllint disable rule:indentation
  - name: 'run some task'
  import_role:
    name: some-role
    # yamllint enable rule:indentation
```
to avoid `warning  comment not indented like content  (comments-indentation)`.

**Comment by adrienverge** [comment]
Could you test with #174? It aims to fix the problem.

**Comment by g1ps** [comment]
Happy to help but how? Is there a build I can install?

**Comment by g1ps** [comment]
Any help here?

**Comment by adrienverge** [comment]
I took time to try to address your problem in #174, would you please test it and give feedback?

A 10-second search on your favorite search engine will tell you how to build a Python package: `python setup.py sdist`. Then install the generated `.tar.gz` file using pip (`pip install --user dist/yamllint-1.15.0.tar.gz`).

**Comment by g1ps** [comment]
I appreciate that but I don't immediately have time to build your code for you. If I can find the time I'll certainly try as I'd like to see it fixed.

**Comment by g1ps** [comment]
I finally managed to give this a go. It built and installed without issue but I see no change in behaviour at all, on Windows. I took some of the example problems I raised here and ran it against them and they all still have the same problems. Comments are still flagged as lacking a space, disabling rules doesn't work.

Thanks.

**Comment by bsolomon1124** [comment]
I see this issue as well when cloning a Git repo on Windows.

I suspected that setting [`newlines: type: dos`](https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.new_lines) would also fix this issue, but it did not.  It seems like those two options would need to work in tandem.

**Comment by bsolomon1124** [comment]
Update: the above does not seem to be working either; wondering if the [`Comment`](https://github.com/adrienverge/yamllint/blob/fec2c2fba736cabf6bee6b5eeb905cab0dc820f6/yamllint/parser.py#L41) class might need modified too.

**Comment by adrienverge** [comment]
I don't have all this in mind anymore, but I'd certainly review a PR :+1: (it needs to come with tests, and more importantly not break existing behavior for other platforms)

About:
```python
elif comment.buffer[text_start] not in (' ', '\n', '\r\n', '\0'):
```
it seemed a good idea, but `comment.buffer[text_start]` is only *one* character, so it will never match `\r\n`. You could try:
```python
elif comment.buffer[text_start] not in (' ', '\n', '\0') and comment.buffer[text_start:text_start+2] != '\r\n':
```
?

**Comment by bsolomon1124** [comment]
This can probably be closed via 512fe17.

**Comment by adrienverge** [comment]
Yes. Thanks for your contribution @bsolomon1124!

**Comment by g1ps** [comment]
Thanks from me, too. I will check it here when I can. Don't hold your breath. ;-)
